crates/services/consensus_module/poa/src/service.rs

```diff
@@ -476,6 +479,7 @@ where
                 if let Some(request) = request {
                     match request {
                         Request::ManualBlocks((block, response)) => {
+                            let _ = self.sync_port.sync_with_peers().await;
```

Comment:
We need to do that on each iteration of the `run` function, not only during manual block production